api/controllers/AuthController.ts

```diff
@@ -65,6 +71,15 @@ export class AuthController {
     return { error: null };
   }
 
+  @UseBefore(UserAuthentication)
+  @Post('/modifyEmail')
+  async modifyEmail(@Body() modifyEmailRequest: ModifyEmailRequest,
+    @AuthenticatedUser() user: UserModel): Promise<ModifyEmailResponse> {
```

Comment:
nit: can we rename this route to `/emailModification`? Even though it doesn't sound as clean as `/modifyEmail`, I think we should be consistent in API naming, since the other /auth routes are all nouns (like `/emailVerification` vs. `/verifyEmail` or `/passwordReset` vs. `/resetPassword` - we changed from verbs to nouns during the TS rewrite)